### ISSUE TITLE
Toning down all possible responses/output

### ISSUE THREAD

**Issue by svpernova09** [memtech/elvis#44]
This may come off a bit as me being the "fun police" but I'm fine with that.

I'd like to propose that anything that isn't PG-13 be limited to the extent it will not output to the #memtech channel or otherwise removed from Elvis. 

Those of us who are in the channel day in and day out "get" the jokes but I'm really concerned with potential new comers and the possibility of elvis outputting offensive images or other potentially offensive content.

TL;DR:
Anything that cannot be strictly limited to a PG-13 rating should be disabled or removed.

**Comment by bradmontgomery** [comment]
:+1: to keeping it pg-13.  Or G for that matter. It's not unusual for me to have kids walk by my computer screen.

**Comment by dpritchett** [comment]
I'm not opposed to this in general, and we could put something in Elvis's docs about not contributing questionable stuff.  Can you give me a list of the things to look into?

Also, we have questionable content generated by real live humans day in and day out - where's the line?  Elvis doesn't even respond without human input at this point; some humans are triggering these questionable responses.

I can see a case for cleaning up the google image results (god knows how, but I get it); but beyond that Elvis is simply enhancing other people's fundamentally NSFW personalities.

I'm asking because we on the Elvis project have a history of bending to these requests and getting little in the way of thanks other than more of the same requests.  As long as you keep an informal, undocumented code of conduct this is never going to end.

**Comment by unstablereality** [comment]
Users having NSFW conversations is one thing, and that's understandable as that is human-entered content.  The bot responding with NSFW material could imply that the channel and #memtech approves of and/or condones of such material.  I think this is a step in the right direction.  We can still have a lot of fun with the bot when it isn't responding with NSFW material.  The only exception I could see is the meme generator, because again that is directly supplied by the user at the moment and not elvis automatically responding to commands with NSFW phrases or images.

**Comment by dpritchett** [comment]
FWIW I disabled the `image me` and `youtube me` scripts at work because they go off the rails so often.

**Comment by unstablereality** [comment]
That makes me sad, but it makes sense and I don't disagree with it if we can't force the results to be more in line with what we want.  Was `animate me` part of `image me`, or do we need to disable that as well?

**Comment by dpritchett** [comment]
Pretty sure `animate me` is just a wrinkle on image me, I'd say we could pull that too.  I can always google for accurate results anyhoo.

**Comment by dpritchett** [comment]
Oh and by "at work" I mean on a separate Hubot entirely.  Haven't done anything to Elvis today, but I'm really open to whatever at this point.  Elvis isn't actually doing anything in other channels, and my personal contributions have dwindled to zero since our last Elvis spat in 2014.

**Comment by speakingcode** [comment]
the best way to ruin a good thing is to make it better.

who are these people that are scared away or pushed off from the scene
because of activity in #memtech(freenode) ? i've yet to hear from or meet a
single one, but i have seen several active, long-standing people of this
community being alienated and/or feeling like they are being alienated, for
the sake of these yet unknown potential newcomers.

if memphis technology foundation has concerns over what dialogue is
associated with it, then register #mtf (or another appropriately named
channel) and let that be MTF's official space with whatever rules and
restrictions are fit per the MTF mission. I do recall that it has been
explicitly stated that MTF will not assume ownership or command of the
#memtech concept and its manifestations as it is public domain property of
the community.

On Mon, Feb 23, 2015 at 10:49 AM, Daniel J. Pritchett <
notifications@github.com> wrote:

> Oh and by "at work" I mean on a separate Hubot entirely. Haven't done
> anything to Elvis today, but I'm really open to whatever at this point.
> Elvis isn't actually doing anything in other channels, and my personal
> contributions have dwindled to zero since our last Elvis spat in 2014.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/memtech/elvis/issues/44#issuecomment-75581016>.
>



-- 
Daniel Lissner
http://speakingcode.com

**Comment by archavius** [comment]
Weren't we just talking to dBoze yesterday about him and a few friends being put off by the banter in #MemTech?

**Comment by speakingcode** [comment]
He's in the channel right now, though, and regularly attends user groups
and other events. I believe he said some things raised his eyebrow
wondering if others might be bothered, but he was not too bothered himself.
Either way, I didn't gather that he was chased off.

On Mon, Feb 23, 2015 at 11:28 AM, archavius <notifications@github.com>
wrote:

> Weren't we just talking to dBoze yesterday about him and a few friends
> being put off by the banter in #MemTech?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/memtech/elvis/issues/44#issuecomment-75589360>.
>



-- 
Daniel Lissner
http://speakingcode.com

**Comment by bradmontgomery** [comment]
imho, #memtech should *police* (for lack of a better word) itself. I'm pretty sure that's what's happening now. Everyone in this thread is part of the community.

*My* preference is that #memtech (and elvis) stay mostly-SFW.

**Comment by archavius** [comment]
Speaking Code: My impression was he showed back up in IRC today because of the dialog we had with him (and his other friends) about his negative experiences in the channel ; which I think is a good thing BTW. I'm newer to the scene so no idea on how long he's been around. Thought it was valid. /shrugs

All good.

**Comment by svpernova09** [comment]
> Can you give me a list of the things to look into?

Off the top of my head: Image results, video results (Looks like you've already addressed this)

I'm also concered about text based responses. We should probably just remove the twss response.

I'm more than happy to go through the code once you've had a chance to once over and offer up anything else for discussion / removal.

> Also, we have questionable content generated by real live humans day in and day out - where's the line?  Elvis doesn't even respond without human input at this point; some humans are triggering these questionable responses.

The human generated questionable content should be dealt with as needed. I'm not anyone's mom or dad. My role in IRC is to try to maintain as much of a welcoming spirit as possible. I'll be the first to admit I can be an asshole and that I'm just as guilty of not maintaining a welcoming spirit all the time. I don't think we have to be perfect all the time. I think we need to recognize that we may be scaring off people by our conversations and attitudes, both from a standpoint of humans and bots.

> I can see a case for cleaning up the google image results (god knows how, but I get it); but beyond that Elvis is simply enhancing other people's fundamentally NSFW personalities.

This request isn't about modifying people's NSFW personalities. This is not the place for THAT discussion. This is a first step in making sure that what **isn't** human output is set to some rating standard.

> I'm asking because we on the Elvis project have a history of bending to these requests and getting little in the way of thanks other than more of the same requests. As long as you keep an informal, undocumented code of conduct this is never going to end.

Some of the previous requests have been due to people coming to me and mentioning they find something offensive. Some of it has been my own two cents. 

I am not, and will never be the benevolent #memtech dictator. It's not my job to tell you how to run your bot.

As a member of the #memtech community it is my responsibility to react when issues are brought to my attention because I am seen as a leader. Just as any of the user group leaders are seen as #memtech leaders. 

What would you as a UG leader do if someone brought something to your attention that made them uncomfortable? You would address it in the best way you can, which is exactly what I'm trying to accomplish here.

I appreciate the fact that the elvis team hosts their repo within the memtech organization. I appreciate the fact it's in a place where these things can be freely discussed and not hidden in some private repo. However if the bot is going to remain welcome in #memtech channel, then in needs to play by the rules of the channel and in line with the wishes of the channel owners. This is common IRC courtesy that dates back since the beginning of IRC culture.

To reemphasize: I'm not dictating terms. I'm trying to address issues that have been brought to me in the best way I can. I'm not the Benevolent Dictator of #memtech, I'm just one of the few people that's willing to do the organizational work to make the community better.

> the best way to ruin a good thing is to make it better.

> who are these people that are scared away or pushed off from the scene because of activity in #memtech(freenode) ? i've yet to hear from or meet a single one, but i have seen several active, long-standing people of this community being alienated and/or feeling like they are being alienated, for
the sake of these yet unknown potential newcomers.

> if memphis technology foundation has concerns over what dialogue is associated with it, then register #mtf (or another appropriately named channel) and let that be MTF's official space with whatever rules and
restrictions are fit per the MTF mission. I do recall that it has been explicitly stated that MTF will not assume ownership or command of the memtech concept and its manifestations as it is public domain property of the community.

This is not coming from, or have anything to do with MTF, this coming from me...the guy that manages the IRC channel. See the section about where I stated "I'm not the Benevolent Dictator of #memtech".

It seems like there is some crossover conversation happening here. This thread is only regarding the Elvis IRC bot, not human personalities. I'm fine with discussing human personallity issues in regards to IRC, but that should be handled on the #memtech google group, not on the github issues of the irc bot.

> imho, #memtech should police (for lack of a better word) itself. I'm pretty sure that's what's happening now. Everyone in this thread is part of the community.

> My preference is that #memtech (and elvis) stay mostly-SFW.

I agree about #memtech policing itself, and that is exactly what this is.

**Comment by sanseref** [comment]
While I'm all for inappropriate stuff, I also want to grow the community. And if we, as the community, are advertising #memtech in our user groups and outreach, I think we need to make sure it's representative of us and have some level of consideration for anyone in the public who joins the channel. Even though I enjoy the exessive banter, I think toning it down is a worthy cause.

**Comment by dpritchett** [comment]
@svpernova09 you can't set down the mantle temporarily.  You are as of this moment the superuser of #memtech IRC, top-level admin of @memtech twitter, and admin of github.com/memtech.  I can't claim any of those three things, and nor can most of the rest of Elvis's list of top contributors.

I'm working on stripping out the image lookups (having trouble with heroku deploys right now though :sob:).

**Comment by svpernova09** [comment]
> @svpernova09 you can't set down the mantle temporarily. You are as of this moment the superuser of #memtech IRC, top-level admin of @memtech twitter, and admin of github.com/memtech. I can't claim any of those three things, and nor can most of the rest of Elvis's list of top contributors.

I am all of those things because no one else stepped up to do it. Are you saying that in order to be one, I can't be the others? None of the titles I hold are relevant to the discussion or fact of "People are complaining about behaviors in IRC"

> I'm working on stripping out the image lookups (having trouble with heroku deploys right now though :sob:).

No wories, this wasn't a 'drop everything and do this' thing.

**Comment by speakingcode** [comment]
Svpernova you make good points and I'm not here to stir trouble; simply
fulfilling my obligation as contrarian ;)

I am personally tolerant of pretty much anything people say, and if it
bothers me personally I simply ignore it (there is a /ignore command in
irc, btw)... regardless I understand most people aren't as desensitized as
I am.

Perhaps I mistook the motivations, or perhaps my view is skewed because I
use a real irc client and I don't have to see images automatically loaded
up in the window. ;-p

I simply brought those points up because the main(only?) ongoing
collaborative tech I've seen produced in memtech besides the channel itself
are Elvis, the webring, and the forum recently set up, each of which have
been under some fire because of how they represent the community - the
community that built them, mind you.

We all know the labors of open source contributors and maintainers and I
just tend to get wary about things like "well hey people can do bad things
and we dont want that so we have to make the tech restrict it" - it's no
easy task, and often is futile effort.

Regardless, I am not here to detract. I think this only opens light on some
other issues. Elvis is entirely a product of and reflection of the current
culture in #memtech. So the real issue is in the culture, and the real
solution is in tactfully fostering an improvement to the culture itself,
which I don't think either conduct codes or coffeescript codes can do much
for. I have ideas/suggestions for what might work but I'll leave that for
request or a separate thread of discussion.

On Mon, Feb 23, 2015 at 11:40 AM, Svpernova09 <notifications@github.com>
wrote:

> Can you give me a list of the things to look into?
>
> Off the top of my head: Image results, video results (Looks like you've
> already addressed this)
>
> I'm also concered about text based responses. We should probably just
> remove the twss response.
>
> I'm more than happy to go through the code once you've had a chance to
> once over and offer up anything else for discussion / removal.
>
> Also, we have questionable content generated by real live humans day in
> and day out - where's the line? Elvis doesn't even respond without human
> input at this point; some humans are triggering these questionable
> responses.
>
> The human generated questionable content should be dealt with as needed.
> I'm not anyone's mom or dad. My role in IRC is to try to maintain as much
> of a welcoming spirit as possible. I'll be the first to admit I can be an
> asshole and that I'm just as guilty of not maintaining a welcoming spirit
> all the time. I don't think we have to be perfect all the time. I think we
> need to recognize that we may be scaring off people by our conversations
> and attitudes, both from a standpoint of humans and bots.
>
> I can see a case for cleaning up the google image results (god knows how,
> but I get it); but beyond that Elvis is simply enhancing other people's
> fundamentally NSFW personalities.
>
> This request isn't about modifying people's NSFW personalities. This is
> not the place for THAT discussion. This is a first step in making sure that
> what *isn't* human output is set to some rating standard.
>
> I'm asking because we on the Elvis project have a history of bending to
> these requests and getting little in the way of thanks other than more of
> the same requests. As long as you keep an informal, undocumented code of
> conduct this is never going to end.
>
> Some of the previous requests have been due to people coming to me and
> mentioning they find something offensive. Some of it has been my own two
> cents.
>
> I am not, and will never be the benevolent #memtech dictator. It's not my
> job to tell you how to run your bot.
>
> As a member of the #memtech community it is my responsibility to react
> when issues are brought to my attention because I am seen as a leader. Just
> as any of the user group leaders are seen as #memtech leaders.
>
> What would you as a UG leader do if someone brought something to your
> attention that made them uncomfortable? You would address it in the best
> way you can, which is exactly what I'm trying to accomplish here.
>
> I appreciate the fact that the elvis team hosts their repo within the
> memtech organization. I appreciate the fact it's in a place where these
> things can be freely discussed and not hidden in some private repo. However
> if the bot is going to remain welcome in #memtech channel, then in needs to
> play by the rules of the channel and in line with the wishes of the channel
> owners. This is common IRC courtesy that dates back since the beginning of
> IRC culture.
>
> To reemphasize: I'm not dictating terms. I'm trying to address issues that
> have been brought to me in the best way I can. I'm not the Benevolent
> Dictator of #memtech, I'm just one of the few people that's willing to do
> the organizational work to make the community better.
>
> the best way to ruin a good thing is to make it better.
>
> who are these people that are scared away or pushed off from the scene
> because of activity in #memtech(freenode) ? i've yet to hear from or meet a
> single one, but i have seen several active, long-standing people of this
> community being alienated and/or feeling like they are being alienated, for
> the sake of these yet unknown potential newcomers.
>
> if memphis technology foundation has concerns over what dialogue is
> associated with it, then register #mtf (or another appropriately named
> channel) and let that be MTF's official space with whatever rules and
> restrictions are fit per the MTF mission. I do recall that it has been
> explicitly stated that MTF will not assume ownership or command of the
> memtech concept and its manifestations as it is public domain property of
> the community.
>
> This is not coming from, or have anything to do with MTF, this coming from
> me...the guy that manages the IRC channel. See the section about where I
> stated "I'm not the Benevolent Dictator of #memtech".
>
> It seems like there is some crossover conversation happening here. This
> thread is only regarding the Elvis IRC bot, not human personalities. I'm
> fine with discussing human personallity issues in regards to IRC, but that
> should be handled on the #memtech google group, not on the github issues of
> the irc bot.
>
> imho, #memtech should police (for lack of a better word) itself. I'm
> pretty sure that's what's happening now. Everyone in this thread is part of
> the community.
>
> My preference is that #memtech (and elvis) stay mostly-SFW.
>
> I agree about #memtech policing itself, and that is exactly what this is.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/memtech/elvis/issues/44#issuecomment-75591842>.
>



-- 
Daniel Lissner
http://speakingcode.com

**Comment by svpernova09** [comment]
@speakingcode 

I'm the same as you in regards to autoloading images, being desensitized. My motivation here is to address the incoming concerns as best I can.

**Comment by dpritchett** [comment]
Closing this PR because the image fix should handle most of it.  If a specific issue comes up please reopen a new issue.  Please continue non-Elvis discussion at https://github.com/memtech/community-guidelines/issues/1
